astropy/modeling/core.py

```diff
@@ -1102,10 +1102,27 @@ def _initialize_parameter_values(self, params):
 
         self._param_metrics = param_metrics
         self._parameters = np.empty(total_size, dtype=np.float64)
+
         # Now set the parameter values (this will also fill
         # self._parameters)
+        # TODO: This is a bit ugly, but easier to deal with than how this was
+        # done previously.  There's still lots of opportunity for refactoring
+        # though, in particular once we move the _get/set_model_value methods
+        # out of Parameter and into Model (renaming them
+        # _get/set_parameter_value)
         for name, value in params.items():
-            setattr(self, name, value)
+            param_descr = getattr(self, name)
+            value = np.array(value)
+            if param_descr._setter is not None:
+                value = param_descr._setter(value)
+            self._parameters[param_metrics[name]['slice']] = value.ravel()
+
+        # Finally validate all the parameters; we do this last so that
+        # validators that depend on one of the other parameters' values will
+        # work
+        for name in params:
+            param_descr = getattr(self, name)
+            param_descr.validator(param_descr.value)
```

Comment:
I think I see what you're saying, and that's a good point.
I think this will be resolved if/when I can get back to working on the new Parameters container interface.  It's a class (similar in intent to one that existed in a much earlier version https://github.com/astropy/astropy/blob/3e6ac2035728a6f247720e266610624efa442597/astropy/modeling/parameters.py#L507 although different in implementation details) that wraps up all book-keeping for parameters in a separate class, so that the Model classes can be made a little simpler (it will also make preparing parameters for passing to Fitters simpler).  The new parameters interface will allow indexing by parameter name, and it will be possible through that interface to update to parameters simultaneously, if needed, by writing
`model.parameters['mu', 'gamma'] = (0.1, 0.25)`, similarly to your example.  So I think that would offer a workaround.